components/datetime/src/options/length.rs

```diff
@@ -211,6 +213,7 @@ pub enum Date {
 /// [`DateTimeFormat`]: super::super::DateTimeFormat
 #[derive(Debug, Clone, Copy, PartialEq)]
 #[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+#[allow(clippy::exhaustive_enums)] // this type is stable
```

Comment:
Nit: This should be `non_exhaustive` because we may want to add additional datetime styles.